core/src/com/unciv/ui/LoadingScreen.kt

```diff
@@ -1,14 +1,30 @@
 package com.unciv.ui
 
+import com.badlogic.gdx.Gdx
+import com.badlogic.gdx.graphics.Pixmap
+import com.badlogic.gdx.graphics.Texture
+import com.badlogic.gdx.graphics.g2d.SpriteBatch
 import com.unciv.Constants
+import com.unciv.ui.popup.Popup
 import com.unciv.ui.utils.BaseScreen
-import com.unciv.ui.utils.extensions.center
-import com.unciv.ui.utils.extensions.toLabel
 
+/** A loading screen that creates a screenshot of the current screen and adds a "Loading..." popup on top of that */
 class LoadingScreen : BaseScreen() {
+    val screenshot = Texture(Pixmap.createFromFrameBuffer(0, 0, Gdx.graphics.backBufferWidth, Gdx.graphics.backBufferHeight))
```

Comment:
NEATO

Reply:
Why is a screenshot even necessary? The effect looks a bit strange - you get the "Loading..." text first in a _wide_ blue rectangle for a while, then in a _narrow_ blue rectangle for another second before the result pops in.